src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -138,14 +139,17 @@ private void handlePublishMessage(ChannelHandlerContext ctx, MqttPublishMessage
         String mqttTopic = publishMessage.variableHeader().topicName();
 
         // perform topic mapping
-        String kafkaMappedTopic = mqttKafkaMapper.map(mqttTopic);
+        MappingResult mqttKafkaMappingResult = mqttKafkaMapper.map(mqttTopic);
+
+        String kafkaMappedTopic = mqttKafkaMappingResult.getKafkaTopic();
+        String key = mqttKafkaMappingResult.getKafkaKey();
```

Comment:
we don't need to save the getter results in two local variables. The Java compiled will do optimization for you. Just call the getter more times when needed.

Reply:
Maybe for more readability?

Reply:
Not sure it's really more readable. But I can understand coming from university they teach it this way ;-)

Reply:
Calling mqttKafkaMappingResult.getKafkaTopic() all the time is kind tiresome for me. 
But if you prefer, I can use mqttKafkaMapper.map(mqttTopic).getKafkaTopic() every I need it.

Reply:
Nope this is wrong, because you are calling the map each time!
I meant calling the map, getting a MappingResult and then calling on it the getters multiple times instead of storing results in variables.

Reply:
Well we are just using kafkaMappedTopic and key two times then I think do not need to store. Thanks!